modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1490,4 +1499,42 @@ protected String getEstimatedExtentFunction() {
                 ? "ST_EstimatedExtent"
                 : "ST_Estimated_Extent";
     }
+
+    public Filter[] splitFilter(Filter filter, SimpleFeatureType schema) {
+        PostPreProcessFilterSplittingVisitor splitter =
+                new PostPreProcessFilterSplittingVisitor(
+                        dataStore.getFilterCapabilities(), schema, null) {
+                    private FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+
+                    @Override
+                    public Object visit(Function expression, Object notUsed) {
+                        if (expression instanceof JsonPointerFunction) {
+                            // takes the json pointer param to check if
+                            // can be encoded
+                            Expression param = expression.getParameters().get(1);
+                            if ((expression.getParameters().get(1) instanceof Literal))
+                                // can encode
+                                fcs.addType(JsonPointerFunction.class);
```

Comment:
Thread safety issue here. The capabilities object is big and normally shared, you should not be modifying it. Since it's large, cloning it is also somehow expensive.
Suggestion: change the PostPreProcessFilterSplittingVisitor so that the "!fcs.supports(expression.getClass())" check is delegated to a separate method that you can override here.

Reply:
fixed